components/nexus-core/src/main/java/org/sonatype/nexus/internal/httpclient/ConnectionEvictionThread.java

```diff
@@ -38,33 +38,34 @@
 
   private final long idleTimeMillis;
 
-  private final long delay;
+  private final long delayMillis;
 
   @VisibleForTesting
   ConnectionEvictionThread(final HttpClientConnectionManager connectionManager,
                            final long idleTimeMillis,
-                           final long delay)
+                           final long delayMillis)
   {
     super("nexus-httpclient-eviction-thread");
     checkArgument(idleTimeMillis > -1, "Keep alive period in milliseconds cannot be negative");
+    checkArgument(delayMillis > 0, "Evicting delay period in milliseconds must be grater than 0");
     this.connectionManager = checkNotNull(connectionManager);
     this.idleTimeMillis = idleTimeMillis;
-    this.delay = delay;
+    this.delayMillis = delayMillis;
     setDaemon(true);
     setPriority(MIN_PRIORITY);
   }
 
-  ConnectionEvictionThread(final HttpClientConnectionManager connectionManager, final Time idleTime) {
-    this(connectionManager, idleTime.toMillisI(), 5000);
+  ConnectionEvictionThread(final HttpClientConnectionManager connectionManager, final Time idleTime, final Time delay) {
+    this(connectionManager, idleTime.toMillisI(), delay.toMillis());
```

Comment:
any reason one argument uses `toMillisI` which coerces the value to an `int` and the other uses `toMillis` which coerces it to a `long`? afaict `ConnectionEvictionThread` expects `long` for both parameters...